open_prices/api/proofs/views.py

```diff
@@ -123,8 +145,20 @@ def upload(self, request: Request) -> Response:
             else settings.ANONYMOUS_USER_ID
         )
         source = get_source_from_request(self.request)
+        save_kwargs = {
```

Comment:
i think i would have done this in the Proof model's `clean` method instead

Reply:
Yes, if you prefer! As this solves one of the biggest duplicate issue, I will merge it now (so that I can deploy it quickly), and let you do the change in a new PR.